go/vt/vtgate/executor_test.go

```diff
@@ -468,6 +468,69 @@ func TestExecutorSet(t *testing.T) {
 		}
 	}
 }
+
+func TestExecutorSetMetadata(t *testing.T) {
+	*vschemaacl.AuthorizedDDLUsers = "%"
+	defer func() {
+		*vschemaacl.AuthorizedDDLUsers = ""
+	}()
+	executor, _, _, _ := createExecutorEnv()
+	session := NewSafeSession(&vtgatepb.Session{TargetString: "@master", Autocommit: true})
+
+	set := "set @@vitess_metadata.app_keyspace_v1= '1'"
+	_, err := executor.Execute(context.Background(), "TestExecute", session, set, nil)
+	if err != nil {
+		t.Errorf("%s error: %v", set, err)
+	}
+
+	show := "show vitess_metadata variables like 'app_keyspace%'"
+	result, err := executor.Execute(context.Background(), "TestExecute", session, show, nil)
+	if err != nil {
+		t.Error(err)
+	}
+
+	want := "1"
+	got := string(result.Rows[0][1].ToString())
+	if got != want {
+		t.Errorf("want migrations %s, result %s", want, got)
+	}
+
+	// Update metadata
+	set = "set @@vitess_metadata.app_keyspace_v2='2'"
+	_, err = executor.Execute(context.Background(), "TestExecute", session, set, nil)
+	if err != nil {
+		t.Errorf("%s error: %v", set, err)
+	}
+
+	gotqr, err := executor.Execute(context.Background(), "TestExecute", session, show, nil)
+	if err != nil {
+		t.Error(err)
+	}
+
+	wantqr := &sqltypes.Result{
+		Fields: buildVarCharFields("Key", "Value"),
+		Rows: [][]sqltypes.Value{
+			buildVarCharRow("app_keyspace_v1", "1"),
+			buildVarCharRow("app_keyspace_v2", "2"),
+		},
+		RowsAffected: 2,
+	}
+
+	if !reflect.DeepEqual(gotqr, wantqr) {
+		t.Errorf("want migrations %v, result %v", wantqr, gotqr)
+	}
+
+	show = "show vitess_metadata variables"
```

Comment:
oh I didn't know the `variables` keyword. I assume this is "standard"?

Reply:
Yes, wanted to keep same syntax

```
SHOW [GLOBAL | SESSION] VARIABLES
    [LIKE 'pattern' | WHERE expr]
```
https://dev.mysql.com/doc/refman/8.0/en/show-variables.html